app/services/user_service.rb

```diff
@@ -0,0 +1,58 @@
+module UserService
+  def self.create(params:)
+    new_user = _user_from_params(params)
+
+    if new_user.performer?
+      User.transaction do
```

Comment:
All of this should probably be wrapped in a transaction, not just this part. If you don't, the `new_user.save!` could fail and leave the old user deactivated without a replacement.